### PR TITLE
Single sign-on with OpenID Connect with Account Management Token

### PR DESCRIPTION
**Pull request by mwan-ep**
Updates SSO OpenID connect how to guide to use account management token.

### CODE REVIEW

File: guides/How-To/Authentication/get-single-sign-on-account-management-token.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+- `ui_locales` - Control the language that the login screen uses.
+
+Commerce forwards these values to the configured Identity Provider, and you should consult your Identity Providers documentation to determine the allowed values.
+
+### Log in to the OpenID Connect provider
+
+The Composable Commerce application should now redirect the user to the modified `authorization-endpoint` link, which begins the single sign-on flow for the user.
+
+### Assign the Account Member to an Account
+If for setting up for the first time the user will end up as an unassigned member. If the account member is already assigned to the account skip over to [Use the authorization code to retrieve an Account Management Authentication Token](#use-the-authorization-code-to-retrieve-an-account-management-authentication-token).
```
Comment (rahinisrinivas) [inline guides/How-To/Authentication/get-single-sign-on-account-management-token.md:160]:
```suggestion
When setting up for the first time, if the user is not yet assigned, they will be  categorized as an unassigned member. If the account member is already assigned, skip over to [Use the authorization code to retrieve an Account Management Authentication Token](#use-the-authorization-code-to-retrieve-an-account-management-authentication-token).
```

File: guides/How-To/Authentication/get-single-sign-on-account-management-token.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+            "type": "account_member",
+            "name": "John Smith",
+            "email": "jsmith@example.com",
+            // Other fields...
+        }
+    ],
+    // Other fields...
+}
+```
+Make a note of the `data[x].id` which will be used as the `account_member_id` when creating an account membership.
```
Comment (rahinisrinivas) [inline guides/How-To/Authentication/get-single-sign-on-account-management-token.md:183]:
```suggestion
Make a note of the `data[x].id`,  which will be used as the `account_member_id` when creating an account membership.
```

File: guides/How-To/Authentication/get-single-sign-on-account-management-token.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    // Other fields...
+}
+
+```
+
+The account member will now be associated to an account.
+
+### Use the authorization code to retrieve an Account Management Authentication Token
+
+After the [Log in to the OpenID Connect provider](#log-in-to-the-openid-connect-provider) step, the user follows a series of redirects to authenticate. Eventually they are redirected back to this Composable Commerce application, with some additional parameters as described in [Successful Authentication Response](https://openid.net/specs/openid-connect-core-1_0.html#AuthResponse).
```
Comment (rahinisrinivas) [inline guides/How-To/Authentication/get-single-sign-on-account-management-token.md:222]:
```suggestion
After the [Log in to the OpenID Connect provider](#log-in-to-the-openid-connect-provider) step, the user follows a series of redirects to authenticate. Eventually, they are redirected back to this Composable Commerce application, with some additional parameters as described in [Successful Authentication Response](https://openid.net/specs/openid-connect-core-1_0.html#AuthResponse).
```

File: guides/How-To/Authentication/get-single-sign-on-account-management-token.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+After validating that the `state` parameter matches the value sent in the [Add parameters to the Authorization Endpoint URL](#add-parameters-to-the-authorization-endpoint-url) section, the Composable Commerce application can use the `code` parameter in the URL, and the `code_verifier` from the [Generate Proof Key for Code Exchange (PKCE) Parameters](#generate-proof-key-for-code-exchange-pkce-parameters) section to get an [Account Management Authentication Token](https://elasticpath.dev/docs/api/accounts/post-v-2-account-members-tokens).
+
+Request an Account Management Authentication Token using the `oidc` as the `authentication_mechanism`. Use `code` generated as the `oauth_authorization_code`, `code_verifier` as `oauth_code_verifier`, and provide the redirect URI of the store which handled the response.
```
Comment (rahinisrinivas) [inline guides/How-To/Authentication/get-single-sign-on-account-management-token.md:226]:
```suggestion
Request an Account Management Authentication Token using the `oidc` as the `authentication_mechanism`. Use `code` generated as the `oauth_authorization_code`, `code_verifier` as `oauth_code_verifier`, and provide the redirect URI of the store, which handled the response.
```

File: guides/How-To/Authentication/get-single-sign-on-account-management-token.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+            "id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+            // Other fields...
+        },
+        // Other ODIC profiles...
+    ],
+}
+
+```
+
+Make note of the `authorization-endpoint` link.
```
Comment (rahinisrinivas) [inline guides/How-To/Authentication/get-single-sign-on-account-management-token.md:111]:
```suggestion
Make a note of the `authorization-endpoint` link.
```

File: guides/How-To/Authentication/get-single-sign-on-account-management-token.md [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+}
+```
+Make a note of the `data[x].id` which will be used as the `account_member_id` when creating an account membership.
+
+#### Request
+```sh
+curl -X POST https://useast.api.elasticpath.com/v2/accounts/:id/account-memberships \
+     -H "Authorization: Bearer XXXX" \
+     -H "Content-Type: application/json" \
+     -d $'{
```
Comment (rahinisrinivas) [inline guides/How-To/Authentication/get-single-sign-on-account-management-token.md:190]:
```suggestion
     -d $ {
```

File: guides/How-To/Authentication/get-single-sign-on-account-management-token.md [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+        "data":{
+          "type": "account_membership",
+          "account_member_id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
+        }
+     }'
```
Comment (rahinisrinivas) [inline guides/How-To/Authentication/get-single-sign-on-account-management-token.md:195]:
```suggestion
     }
```

File: guides/How-To/Authentication/get-single-sign-on-account-management-token.md (hunk 6 above)
Comment (mwan-ep) [reply guides/How-To/Authentication/get-single-sign-on-account-management-token.md:190]: Not sure why we want to remove the quote here. We need these for the CURL call. other requests in the how to guides use single quotes here.

Comment (rahinisrinivas) [reply guides/How-To/Authentication/get-single-sign-on-account-management-token.md:190]: I thought it's not needed. It doesn't show in postman, does it?

Comment (mwan-ep) [reply guides/How-To/Authentication/get-single-sign-on-account-management-token.md:190]: Postman does do it a little differently than what we have here. But anyways I tested without the single quote there and it breaks.

Comment (rahinisrinivas) [reply guides/How-To/Authentication/get-single-sign-on-account-management-token.md:190]: okay. Thank you for testing it in postman.  Please ignore my suggestion.

**Review by rahinisrinivas — APPROVED** [review]

**Review by davidmoltin — APPROVED** [review]

**Pull request merged**